### PR TITLE
Infographic table of contents

### LINKED ISSUE #659: Add table of contents links to infographics
UX participants felt the "on this page" feature, which was included in the Testing wireframes, was helpful to understand what was exactly on the page before scrolling through it all. Nearly all participants attempted to navigate via this table of contents.

The idea is to a list every panel at the top of each infographic so users can click, and quickly navigate to the desired graphic. Consider this addition, but weigh it against the page getting too cluttered.

### PR DESCRIPTION
**Pull request by GarvinH**
closes #659

### CODE REVIEW

**Review by ktw1016 — CHANGES_REQUESTED** [review]
This concept is good to me, I was kinda hoping I'd somehow incorporate this functionality with panel filtering (which I'm not sure how that'd be possible) but this is more realistic solution. 
- [x] font size of "Table of contents" is inconsistent with  "Filter panels". I think they can both be the smaller size of the two
- [x] On panel filter, click on "All" or "None". It scrolls to the bottom of the page, which shouldn't happen
- [x] Table of contents panel list not updating well with panel filtering. For example, on finance infograph, click on "Transfer payments by recipient region". Table of contents show correctly 1 panel. Then click on anything else on panel filter. Table of contents not updating

**Comment by GarvinH** [comment]
I'm not able to replicate the panel filter scrolling to the bottom

**Comment by GarvinH** [comment]
- [ ] When all/none selected, only 9 of 10 links show

the tp_by_region loads really slow -> slow enough for the table of contents to not be able to retrieve a title

**Comment by AlexCLeduc** [comment]
> I'm not able to replicate the panel filter scrolling to the bottom

If you get a panel key in the URL (e.g. by using the ToC) and then filter panels, each time you change a filter, if the panel identified by the URL exists, it will scroll to it. I think this is probably a bug with the filter not clearing the URL parameter, nothing to do with your ToC

**Comment by ktw1016** [comment]
Ah I see, cool!

**Review by ktw1016 — COMMENTED** [review]
2 minor comments, and I'm pushing a commit to showcase a different way to get the titles..

I'm not really a fan of the boilerplate code having to manage `visible_panel_title` state with `add_visible_panel_title`, having to pass that down to panel component code. I think doing it in the proposed way is cleaner, gives more control of the panel titles at the infographic level. It will definitely be more work as we'll need to change how title gets passed down for all panels but that's shouldn't be a biggie

Essentially, it's passing down the `title` at a higher level on `PanelRegistry` which then, `Infographic` can access the panel objects to access `title`. Example with 3 panels. Let me know if that makes sense, and let me know your opinions on which approach is better.

**Comment by ktw1016** [comment]
Oh, one more thing, panel-inventory route is broken

File: client/src/infographic/Infographic.js [hunk 1]
```diff
@@ -234,17 +256,29 @@ class InfoGraph_ extends React.Component {
...
                 set_panel_filter={(panel_filter) =>
-                  this.setState({ panel_filter })
+                  this.setState({ panel_filter, mounted: false })
                 }
               />
             )}
+
+          <TableOfContents
+            loading={loading}
+            mounted={mounted}
```
[the remark this answers is not in the document]
Comment (GarvinH) [reply client/src/infographic/Infographic.js:280]: Whoops I was removing the internal use for testing purposes but forgot to remove it from here. thanks!

**Comment by AlexCLeduc** [comment]
This is pretty cool! You can use "Table des matières"  and "Passer directement à un panneau" for the french

**Review by ktw1016 — APPROVED** [review]
Awesome👍 Super minor comments, otherwise good to me

**Pull request merged**